### PR TITLE
Fix Typo in FindPytorch.cmake

### PR DESCRIPTION
**Pull request by Magnus1cheng**
## Description

Fixes typo in FindPytorch.cmake. REGEX could not match for ROCm error message.

### CODE REVIEW

**Comment by Magnus1cheng** [comment]
or we can use this way to match both strings.
`string(REGEX REPLACE "(No CUDA runtime[^\n]*\n)|(No ROCm runtime[^\n]*\n)" "" Pytorch_CUDA "${Pytorch_CUDA}")`
`string(REGEX REPLACE "(No CUDA runtime[^\n]*\n)|(No ROCm runtime[^\n]*\n)" "" Pytorch_ROCM "${Pytorch_ROCM}")`

**Comment by maxhgerlach** [comment]
Thank you for the contribution, @Magnus1cheng! Please note that you would need to sign off your commits (DCO), see https://github.com/horovod/horovod/pull/3593/checks?check_run_id=7214166372 for some pointers.

> or we can use this way to match both strings.
> ```
> string(REGEX REPLACE "(No CUDA runtime[^\n]*\n)|(No ROCm runtime[^\n]*\n)" "" Pytorch_CUDA "${Pytorch_CUDA}")a
> string(REGEX REPLACE "(No CUDA runtime[^\n]*\n)|(No ROCm runtime[^\n]*\n)" "" Pytorch_ROCM "${Pytorch_ROCM}")
> ```

I'm not sure under which conditions these error messages appear. If either could show up here, it might make sense to filter both in any case.

There's also this second spot: https://github.com/horovod/horovod/blob/9d56c5a6d55e9619e100d19480a11952d116f4b3/cmake/Modules/FindPytorch.cmake#L38

Should that filter also be extended?

**Comment by Magnus1cheng** [comment]
Hi, @maxhgerlach, I updated my commit. This cause error on my environment before so I match both Regex by using my commit, and the error gone. Somehow I cannot reproduce it now.
And for line 38, yes, we may need to change it as well, but I didn't encounter error in line 38, but I guess that changing would not harm anything because it includes the original regex match logic, but extend it. So you decide if we need that line changing.

**Comment by EnricoMi** [comment]
Instead of `"(No CUDA runtime[^\n]*\n)|(No ROCm runtime[^\n]*\n)"`,
why not `"No (CUDA|ROCm) runtime[^\n]*\n"`?

**Comment by Magnus1cheng** [comment]
Yes, that's better. Modified.

File: cmake/Modules/FindPytorch.cmake [hunk 1]
```diff
@@ -24,18 +24,18 @@ endif()
 
 execute_process(COMMAND ${PY_EXE} -c "import torch; from torch.utils.cpp_extension import CUDA_HOME; print(True if ((torch.version.cuda is not None) and (CUDA_HOME is not None)) else False)"
                 OUTPUT_VARIABLE Pytorch_CUDA OUTPUT_STRIP_TRAILING_WHITESPACE)
-string(REGEX REPLACE "No CUDA runtime[^\n]*\n?" "" Pytorch_CUDA "${Pytorch_CUDA}")
+string(REGEX REPLACE "(No (CUDA|ROCm) runtime[^\n]*\n)" "" Pytorch_CUDA "${Pytorch_CUDA}")
```
Comment (EnricoMi) [inline cmake/Modules/FindPytorch.cmake:27]:
```suggestion
string(REGEX REPLACE "No (CUDA|ROCm) runtime[^\n]*\n" "" Pytorch_CUDA "${Pytorch_CUDA}")
```
These brackets are not needed, I think.

Comment (Magnus1cheng) [reply cmake/Modules/FindPytorch.cmake]: Updated

**Review by maxhgerlach — APPROVED** [review]
Thanks for the updates! The changes look good to me.

**Pull request merged**